spacy_llm/pipeline/llm.py

```diff
@@ -117,9 +117,10 @@ def __init__(
         self._name = name
         self._task = task
         self._backend = backend
+        self._save_io = save_io
         self._cache = cache
         self._cache.vocab = vocab
-        self._save_io = save_io
+        self._cache.prompt_template = self._task.prompt_template
```

Comment:
Hmm I don't really like this setup where we override some attributes of the cache but also allow it to be provided on init. This sort of hidden override is hard to reason about. I missed it when we first did this for the vocab.

Reply:
The issue here is that we don't have this information (vocab and prompt template) on cache init time. I'm not a fan of doing it this way either, but don't see another way to do this.

> Hmm I don't really like this setup where we override some attributes of the cache but also allow it to be provided on init.

Neither `prompt_template` nor `vocab` can be provided on init.

Reply:
Ok I see what you mean here. Could we maybe add an `initialize` method like:

```
def initialize(self, vocab: Vocab, task: LLMTask) -> None:
     ...
```

It's kind of a nit but this code path really seems like it could easily lead to bugs down the road

Reply:
Like in https://github.com/explosion/spacy-llm/pull/192/commits/62ec06bd234ead286610fd84293fadbd6f58e821?

Reply:
Exactly yeah, looks good to me